src/Functions/UniqTheta/FunctionsUniqTheta.h

```diff
@@ -0,0 +1,163 @@
+#include <DataTypes/DataTypeAggregateFunction.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <Interpreters/castColumn.h>
+#include <Common/assert_cast.h>
+#include <Common/typeid_cast.h>
+
+#include <AggregateFunctions/AggregateFunctionUniq.h>
+
+namespace DB
+{
+
+
+    namespace ErrorCodes
+    {
+        extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    }
+
+    struct UniqThetaIntersectImpl
+    {
+        static void apply(AggregateFunctionUniqThetaData & sketch_data_1, const AggregateFunctionUniqThetaData & sketch_data_2)
+        {
+            sketch_data_1.set.intersect(sketch_data_2.set);
+        }
+    };
+
+    struct UniqThetaUnionImpl
+    {
+        static void apply(AggregateFunctionUniqThetaData & sketch_data_1, const AggregateFunctionUniqThetaData & sketch_data_2)
+        {
+            sketch_data_1.set.merge(sketch_data_2.set);
+        }
+    };
+
+    struct UniqThetaNotImpl
+    {
+        static void apply(AggregateFunctionUniqThetaData & sketch_data_1, const AggregateFunctionUniqThetaData & sketch_data_2)
+        {
+            sketch_data_1.set.a_not_b(sketch_data_2.set);
+        }
+    };
+
+    template <typename Impl, typename Name>
+    class FunctionUniqTheta : public IFunction
+    {
+    public:
+        static constexpr auto name = Name::name;
+
+        static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionUniqTheta>(); }
+
+        String getName() const override { return name; }
+
+        bool isVariadic() const override { return false; }
+
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+        size_t getNumberOfArguments() const override { return 2; }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+            const auto * sketch_type0 = typeid_cast<const DataTypeAggregateFunction *>(arguments[0].get());
+            if (!(sketch_type0 && sketch_type0->getFunctionName() == "uniqTheta"))
+                throw Exception(
+                        "First argument for function " + getName() + " must be a uniqTheta but it has type " + arguments[0]->getName(),
+                        ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+            const auto * sketch_type1 = typeid_cast<const DataTypeAggregateFunction *>(arguments[1].get());
+            if (!(sketch_type1 && sketch_type1->getFunctionName() == "uniqTheta"))
+                throw Exception(
+                        "Second argument for function " + getName() + " must be a uniqTheta but it has type " + arguments[1]->getName(),
+                        ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+            //todo:check
+            if (sketch_type0->getArgumentsDataTypes()[0]->getTypeId() != sketch_type1->getArgumentsDataTypes()[0]->getTypeId())
```

Comment:
`DataType` class has method `equals`. And also, why do we check only the first argument?

Reply:
@pzhdfy

Reply:
update.   check all ArgumentsDataTypes and use equals